specs/hotel_spec.rb

```diff
@@ -0,0 +1,166 @@
+      @hotel.make_reservation(2222, "2012/12/12", "2012/12/15")
+      @hotel.reservations.reservation_list.length.must_equal 1
+    end #call class
+
+    it "Will Raise an error if asked to reserve an unavailable room" do
+      @hotel.make_reservation(1000, "2001/1/30", "2001/2/4", room: 1)
+      proc {
+        @hotel.make_reservation(1000, "2001/1/30", "2001/2/4", room: 1)
+      }.must_raise ArgumentError
+    end
+  end #make reservations
+
+  describe "Raise Errors for Invalid Dates" do
+    it "will raise an error when an invalid date (check_in or check_out) range is provided" do
+      proc {
+        @hotel.make_reservation(1000, "2001/1/32", "2001/2/4", room: 1)
+      }.must_raise ArgumentError
+
+      proc {
+        @hotel.make_reservation(1000, "2001/1/30", "2001/1/33", room: 1)
+      }.must_raise ArgumentError
+    end #invalid check in
+
+    it "will raise an error if the Check Out Date is earlier than the Check In date" do
+      proc {
+        @hotel.make_reservation(1000, "2001/1/10", "2001/1/5", room: 1)
+      }.must_raise ArgumentError
+    end
+  end #invalid dates
+
+
+  describe "Find Reservations" do
+    it "Can find and return a single reservation for a specific date: " do
+      @hotel.make_reservation(2222, "2012/12/12", "2012/12/15")
+      found = @hotel.find_reservations("2012/12/12", "2012/12/13")
+      found[0].view_reservation.must_equal "ID: 2222, Room: 1, Check in: 2012-12-12, Check Out: 2012-12-15, Total Nights: 3, Total Cost: 600"
+    end #find a single reservation for a specific date
+
+    it "Can find and return multiple reservations for the same date" do
+      @hotel.make_reservation(1203, "2017/12/12", "2017/12/15")
+      @hotel.make_reservation(1204, "2017/12/14", "2017/12/16")
+      found = @hotel.find_reservations("2017/12/14", "2017/12/15")
+      found.length.must_equal 2
+      found[0].id.must_equal 1203
+      found[1].id.must_equal 1204
+    end #multiple reservations
+
+    it "Can return an array of open rooms for a particular date range: " do
+      @hotel.make_reservation(2222, "2012/12/12", "2012/12/15")
+      @hotel.make_reservation(2224, "2012/12/13", "2012/12/17")
+      @hotel.open_rooms("2012/12/12", "2012/12/13").must_be_kind_of Array
+      @hotel.open_rooms("2012/12/12", "2012/12/13")[0].must_equal 3
+    end #open rooms
+  end #find reservations
+
+  describe "Total Cost" do
+    it "Given an ID number it can find the Total Cost of a reservation: " do
+      @hotel.make_reservation(2222, "2012/12/12", "2012/12/15")
+      @hotel.total_cost(2222).must_equal 600
+    end #ID Number
+
+    it "Returns 'Reservation Not Found' if it can't find the ID number: " do
+      @hotel.make_reservation(2222, "2012/12/12", "2012/12/15")
+      @hotel.total_cost(2223).must_equal "Reservation Not Found"
+    end #Not Found
+  end #Total Cost
+
+  describe "Blocks" do
+
+    it "Can call Block to create an array to store block information" do
```

Comment:
Test coverage looks pretty good around blocks. There are just a couple of places where it seems like things are missing:
- What happens if you try to create a block when there are not enough rooms available?
- What if the dates on the reservation don't match those on the block?